typing/typemod.ml

```diff
@@ -707,6 +708,77 @@ let merge_constraint initial_env loc sg lid constr =
   with Includemod.Error explanation ->
     raise(Error(loc, initial_env, With_mismatch(lid.txt, explanation)))
 
+(* A simplified version of [merge_constraint] for the case of packages. Package
+   constraints are much simpler - they must be "with type" constraints for types
+   with no parameters and can only update abstract types. *)
+let merge_package_constraint initial_env loc sg lid cty =
+  let rec patch_item namelist outer_sig_env sg_for_env ~ghosts item =
+    let return replace_by =
+      Some ((), {Signature_group.ghosts; replace_by})
```

Comment:
Since we are by construction not in the destructive substitution case of the signature patching , maybe:
```ocaml
let return replace_by =
      Some ((), {Signature_group.ghosts; replace_by=Some replace_by})
```
?

Reply:
Good point - I agree that is clearer, and have changed it.